### DISCUSSION TITLE
Handling dependencies for internal packages.

### DISCUSSION THREAD

**Discussion by onlit** [vercel/turborepo#9115] [Help] [answered]
### Summary

I have started building microfrontends with next.js and turbo repo. I have a few internal packages created that depend on libraries like redux, tanstack react query, or material ui. So far, everything is functional or so I think.

I'm unsure whether to put these dependencies in the internal package's package.json > dependencies or package.json > peerDependencies.

What's the best practice? Would using dependencies or peerDependencies make any difference?

Here's an example package.json of my ui internal package.
```
{
  "name": "@repo/ui",
  "version": "1.0.0",
  "private": true,
  "exports": {
    "./AdminLayout": "./src/AdminLayout/AdminLayout.tsx",
    "./Inputs/Checkbox": "./src/Inputs/CheckboxField/CheckboxField.tsx",
    "./SearchBar": "./src/SearchBar/SearchBar.tsx",
    "./DataTableTextInput": "./src/DataTableTextInput/DataTableTextInput.tsx",
    "./DataTable": "./src/DataTable/DataTable.tsx",
    "./Auth/*": "./src/Auth/*.tsx"
  },
  "scripts": {
    "lint": "eslint . --max-warnings 0",
    "generate:component": "turbo gen react-component"
  },
  "devDependencies": {
    "@repo/hooks": "workspace:*",
    "@repo/eslint-config": "workspace:*",
    "@repo/typescript-config": "workspace:*",
    "@turbo/gen": "^1.12.4",
    "@types/node": "^20.11.24",
    "@types/eslint": "^8.56.5",
    "@types/react": "^18.2.61",
    "@types/react-dom": "^18.2.19",
    "eslint": "^8.57.0",
    "typescript": "^5.3.3"
  },
  "dependencies": {
    "react": "18.3.1",
    "@tanstack/react-table": "^8.10.7",
    "next": "14.2.1",
    "usehooks-ts": "^2.9.1",
    "formik": "^2.2.9",
    "mui-tel-input": "^4.0.0",
    "@ant-design/icons": "^5.3.6",
    "@emotion/cache": "^11.11.0",
    "@emotion/react": "^11.11.4",
    "@emotion/server": "^11.11.0",
    "@emotion/styled": "^11.11.5",
    "@mui/icons-material": "^5.15.15",
    "@mui/material": "^5.15.15",
    "@mui/material-nextjs": "^5.15.11"
  }
}
```

**Comment by anthonyshew** [comment, marked as answer]
In general, `dependencies` are things that run in production and `devDependencies` are things that you only need while developing.

That said, you may care more about this distinction if:
1. You're publishing your packages. Using `devDependencies` will create smaller package downloads.
2. You want to use the `--production` flag with your package manager to have faster install times in CI.
